### PR TITLE
Add project setup

### PR DESCRIPTION
**Pull request by brinxmat**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    }
+
+    check.dependsOn jacocoTestCoverageVerification
+    jacocoTestCoverageVerification.dependsOn(jacocoTestReport)
+
+    jacocoTestCoverageVerification {
+        violationRules {
+            rule {
+                excludes = ['**datacite**']
```
Comment (brinxmat) [inline build.gradle:77]: To ignore generated classes

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                }
+            }
+
+            rule {
+                limit {
+                    counter = 'CLASS'
+                    value = 'COVEREDRATIO'
+                    minimum = 1.00
+                }
+                excludes = ['**datacite**']
```
Comment (brinxmat) [inline build.gradle]: To ignore generated classess

File: config/pmd/ruleset.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  name="PMD Rules"
+  xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+  xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+  <description>
+    PMD rules
+  </description>
+  <exclude-pattern>.*/test/.*</exclude-pattern>
+  <exclude-pattern>.*/datacite-data/.*</exclude-pattern>
```
Comment (brinxmat) [inline config/pmd/ruleset.xml:11]: We don't want PMD to look at generated datacite classes

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,79 @@
+allprojects {
```
Comment (axthosarouris) [inline build.gradle:5]: I am not sure if you need to add the following element. Probably not :)
```
plugins {
    id 'java'
    id 'jacoco'
    id 'checkstyle'
    id 'pmd'
    id "nebula.lint" version "16.9.0"

}
```

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    repositories {
+        jcenter()
+    }
+    group 'no.unit.nva'
+    version '1.0-SNAPSHOT'
+
```
Comment (axthosarouris) [inline build.gradle:9]: maybe add the compatibility claims, just to be sure that we are always referring to the correct version

```

sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11
```

File: config/checkstyle/checkstyle.xml [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+      <property name="allowMissingParamTags" value="true"/>
+      <property name="allowMissingReturnTag" value="true"/>
+      <property name="allowedAnnotations" value="Override, Test"/>
+      <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF"/>
+    </module>
+    <module name="MissingJavadocMethod">
+      <property name="scope" value="public"/>
+      <property name="minLineCount" value="2"/>
+      <property name="allowedAnnotations" value="Override, Test"/>
+      <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF"/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle.xml]: remove CTOR_DEF from here. We do not need to add Javadoc to all constructors.

File: config/checkstyle/checkstyle.xml [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+      <property name="scope" value="public"/>
+      <property name="allowMissingParamTags" value="true"/>
+      <property name="allowMissingReturnTag" value="true"/>
+      <property name="allowedAnnotations" value="Override, Test"/>
+      <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF"/>
+    </module>
+    <module name="MissingJavadocMethod">
+      <property name="scope" value="public"/>
+      <property name="minLineCount" value="2"/>
+      <property name="allowedAnnotations" value="Override, Test"/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle.xml]: Add ParametrizedTest to avoid having to add Javadoc to all ParameterizedTests

**Review by axthosarouris — APPROVED** [review]

**Review by axthosarouris — CHANGES_REQUESTED** [review]
This is the only requested change. The previous are comments: travis is missing

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**